src/methods/properties.js

```diff
@@ -501,6 +501,12 @@ export class PropertiesImplementation {
       let key = InternalGetPropertiesKey(P);
       let map = InternalGetPropertiesMap(O, P);
       let propertyBinding = map.get(key);
+      if (propertyBinding === undefined && O.isPartialObject()) {
+        // if we try and delete a property that we aren't sure exists
+        // throw a FatalError rather than an invariant
+        AbstractValue.reportIntrospectionError(O, P);
```

Comment:
This is a deprecated way of generating diagnostic messages. The modern way is to always do so in a context where we know the source location and to always have a dedicated message and code for every distinct kind of problem.

Reply:
Can you link me to an example of the sort of error I should be using here? I noticed @sebmarkbage was using this pattern in `CallExpression.js`, so I just re-used that.

Reply:
Look for places where CompilerDiagnostic is created in CallExpression.js.